integration_tests/test.sh

```diff
@@ -63,6 +65,15 @@ fi
 status=0
 failures=""
 echo "Doing schema validation..."
+
+if ! [ -d "${INTEGRATION_TEST_VENV}" ]; then
```

Comment:
You are testing that `${INTEGRATION_TEST_VENV}` is not a directory here...  You probably want a positive test for a regular file here, e.g.: `if [ -f "${INTEGRATION_TEST_VENV}" ]; then`

A negative test for 'not a directory' will test true if for some reason ${INTEGRATION_TEST_VENV} gets unset (`""`), and will also pass if it is a pipe, device file, etc..

Reply:
The idea was to only do `virtualenv` (and the `pip install`) if the dir didn't exist -- so I don't think `test -f` does what I want, either, since that also returns false when it doesn't exist.

I guess `test -x ${VENV}/bin/python` is probably safer than either.

It feels like a natural fit for the Makefile, but I left it in here in keeping with the ability to run `test.sh` on its own.

Reply:
Ah, you are right - I read the intent wrong.  Carry on :)